tests/callbacks/test_finetuning_callback.py

```diff
@@ -365,3 +365,118 @@ def forward(self, x):
     # conv0.weight, conv0.bias, bn0.weight, bn0.bias, parent_param
     # conv1.weight, conv1.bias, bn1.weight, bn1.bias
     assert len(encoder_params) == 9
+
+
+class TestCallbacksRestoreCallback(BaseFinetuning):
+
+    def freeze_before_training(self, pl_module):
+        self.freeze(pl_module.layer[:3])
+
+    def finetune_function(self, pl_module, epoch, optimizer, opt_idx):
+        if epoch >= 1:
+            self.unfreeze_and_add_param_group(pl_module.layer[epoch - 1], optimizer)
+
+
+class FinetuningBoringModel(BoringModel):
+
+    def __init__(self):
+        super().__init__()
+        self.layer = nn.Sequential(nn.Linear(32, 32), nn.Linear(32, 32), nn.Linear(32, 32), nn.Linear(32, 2))
+
+    def configure_optimizers(self):
+        parameters = filter(lambda x: x.requires_grad, self.parameters())
+        optimizer = torch.optim.SGD(parameters, lr=0.1)
+        return optimizer
+
+
+def test_callbacks_restore(tmpdir):
+    """
+    Test callbacks restore is called after optimizers have been re-created
+    but before optimizer states reload
+    """
+
+    seed_everything(42)
```

Comment:
Is the fixed seed strictly necessary for the test?

Reply:
addressed all your comments @carmocca